commit 095a2aa48fc6009d2a3ccc57e8543dc40d43e509
Author: Shaun Reich <shaun.reich@kdemail.net>
Date:   Fri May 4 16:54:36 2012 -0400

    fix build errors with CLANG/LLVM
    (cherry picked from commit ac48ea95c7f726b7208b6b6136d97beefe7b06d9)

diff --git a/nepomuk/services/storage/datamanagementmodel.h b/nepomuk/services/storage/datamanagementmodel.h
index 781fb2c..5d9cbf1 100644
--- nepomuk/services/storage/datamanagementmodel.h
+++ nepomuk/services/storage/datamanagementmodel.h
@@ -156,7 +156,7 @@ public Q_SLOTS:
                         const QString& app,
                         Nepomuk::StoreIdentificationMode identificationMode = Nepomuk::IdentifyNew,
                         Nepomuk::StoreResourcesFlags flags = Nepomuk::NoStoreResourcesFlags,
-                        const QHash<QUrl, QVariant>& additionalMetadata = QHash<QUrl, QVariant>() );
+                        const QHash<QUrl, QVariant>& additionalMetadata = (QHash<QUrl, QVariant>()));
 
     /**
      * Merges two resources into one. Properties from \p resource1
@@ -188,7 +188,7 @@ public Q_SLOTS:
                          const QString& userSerialization = QString(),
                          Nepomuk::StoreIdentificationMode identificationMode = Nepomuk::IdentifyNew,
                          Nepomuk::StoreResourcesFlags flags = Nepomuk::NoStoreResourcesFlags,
-                         const QHash<QUrl, QVariant>& additionalMetadata = QHash<QUrl, QVariant>());
+                         const QHash<QUrl, QVariant>& additionalMetadata = (QHash<QUrl, QVariant>()));
 
     /**
      * Describe a set of resources, i.e. retrieve all their properties.
@@ -204,7 +204,7 @@ public Q_SLOTS:
     //@}
 
 private:
-    QUrl createGraph(const QString& app = QString(), const QHash<QUrl, QVariant>& additionalMetadata = QHash<QUrl, QVariant>());
+    QUrl createGraph(const QString& app = QString(), const QHash<QUrl, QVariant>& additionalMetadata = (QHash<QUrl, QVariant>()));
     QUrl createGraph(const QString& app, const QMultiHash<QUrl, Soprano::Node>& additionalMetadata);
 
     /**
diff --git a/nepomuk/services/storage/resourcemerger.cpp b/nepomuk/services/storage/resourcemerger.cpp
index e6f352c..1d49e65 100644
--- nepomuk/services/storage/resourcemerger.cpp
+++ nepomuk/services/storage/resourcemerger.cpp
@@ -506,7 +506,7 @@ Soprano::Node Nepomuk::ResourceMerger::resolveUnmappedNode(const Soprano::Node&
     m_mappings.insert( QUrl(node.toN3()), newUri );
 
     // FIXME: trueg: IMHO these statements should instead be added to the list of all statements so there is only one place where anything is actually added to the model
-    Soprano::Node dateTime( Soprano::LiteralValue( QDateTime::currentDateTime() ) );
+    Soprano::Node dateTime = Soprano::LiteralValue(QDateTime::currentDateTime());
     m_model->addStatement( newUri, NAO::created(), dateTime, m_graph );
     m_model->addStatement( newUri, NAO::lastModified(), dateTime, m_graph );
 
